plugins/integration/providers/voicebase/lib/model/IntegrationVoicebaseProviderValidator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * @package plugins.voicebase
+ * @subpackage model
+ */
+class IntegrationVoicebaseProviderValidator implements IIntegrationProvider
+{
+	/* (non-PHPdoc)
+	 * @see IIntegrationProvider::getPermissions($partnerId)
+	 */
+	public static function getPermissions($partnerId)
+	{
+		$permissionNames = array(VoicebasePlugin::PARTNER_LEVEL_PERMISSION_NAME);
+		return PermissionPeer::retrievePartnerLevelPermissions($partnerId, null, $permissionNames);
```

Comment:
You could use PermissionPeer::isAllowedPlugin or isValidForPartner instead of changing existing method.